packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,202 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+// https://github.com/prisma/prisma/issues/8832
+testMatrix.setupTestSuite(
+  ({ provider }) => {
+    async function createTags(length: number): Promise<number[]> {
+      const ids = Array.from({ length }, (_, i) => i + 1)
+      const data = ids.map((id) => ({ id }))
+      await prisma.tag.createMany({
+        data,
+      })
+      return ids
+    }
+
+    async function getTagsFindManyIn(ids: number[]): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        where: {
+          id: {
+            in: ids,
+          },
+        },
+      })
+      return tags
+    }
+
+    async function getTagsFindManyInclude(): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        include: {
+          posts: true,
+        },
+      })
+      return tags
+    }
+
+    async function clean() {
+      const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+      await prisma.$transaction(cleanPrismaPromises)
+    }
+
+    describe('issue #8832', () => {
+      beforeEach(async () => {
+        await clean()
+      }, 10_000)
+
+      test('should succeed when "in" has 32766 ids', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32766 records', async () => {
+        const n = 32766
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32767 ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
```

Comment:
Why do we need all these tests (32766, 32767, 32768)?
Can we only keep the last 2 with n = 32768 maybe?

Reply:
Or remove them all and only keep the ones below where n = 65536?

Reply:
Because previously the ones with `32767` and `32768` were failing. We shouldn't test a single `n` because there's also a query batching logic kicking in around those numbers (at a certain point, it used to work with `32766` and `32768` records, but not with `32767` records).